pyquil/quilbase.py

```diff
@@ -145,18 +145,22 @@ def _convert_to_rs_instructions(instrs: Iterable[AbstractInstruction]) -> List[q
 
 
 def _convert_to_py_instruction(instr: quil_rs.Instruction) -> AbstractInstruction:
-    if not isinstance(instr, quil_rs.Instruction):
-        raise ValueError(f"{type(instr)} is not a valid Instruction type")
-    if instr.is_calibration_definition():
-        return DefCalibration._from_rs_calibration(instr.to_calibration_definition())
-    if instr.is_gate():
-        return Gate._from_rs_gate(instr.to_gate())
-    if instr.is_measure_calibration_definition():
-        return DefMeasureCalibration._from_rs_measure_calibration_definition(instr.to_measure_calibration_definition())
-    if instr.is_measurement():
-        return Measurement._from_rs_measurement(instr.to_measurement())
+    if isinstance(instr, quil_rs.Instruction):
```

Comment:
Sorry for swooping in here, but:

1. Is `quil_rs.Instruction` a "superclass" (Rust stuff etc. etc.) of the other types here?
2. Do we want a serial run of `if` statements, or should some of these be `elif`s?

Reply:
Not at all, thanks for reviewing!

1. Sort of. `quil_rs.Instruction` is a binding to an `Enum` of the same name in `quil-rs`. It would be really nice if we could use it on the Python side as a true superclass, similar to `AbstractInstruction`, but `pyo3` doesn't allow you to subclass enums. So we have to do some gymnastics here to workaround that fact. This may be something we circle back to later by playing with wrapper types, or converting `quil-py` to a hybrid Rust/Python crate so we can use Python local to the package to insulate users from some of this. If you are interested, [this is the API for `quil_rs.Instruction`](https://github.com/rigetti/quil-rs/blob/1455-python-support-for-program-and-beyond/quil-py/quil/instructions/__init__.pyi#L7).

2. Each branch returns early, so `if` and `elif` should be equivalent in this case, though I'm open to stylistic arguments to switching them.

Reply:
Hadn't noticed the early returns, sorry! Thanks for the response